sqlmesh/utils/date.py

```diff
@@ -353,16 +353,16 @@ def to_time_column(
         return exp.cast(time_column, to=time_column_type)
     if time_column_type.is_type(exp.DataType.Type.DATE):
         return exp.cast(exp.Literal.string(to_ds(time_column)), to="date")
-    if time_column_type.this in TEMPORAL_TZ_TYPES:
-        return exp.cast(exp.Literal.string(to_tstz(time_column)), to=time_column_type.this)
-    if time_column_type.this in exp.DataType.TEMPORAL_TYPES:
-        return exp.cast(exp.Literal.string(to_ts(time_column)), to=time_column_type.this)
+    if time_column_type.is_type(*TEMPORAL_TZ_TYPES):
+        return exp.cast(exp.Literal.string(to_tstz(time_column)), to=time_column_type)
+    if time_column_type.is_type(*exp.DataType.TEMPORAL_TYPES):
+        return exp.cast(exp.Literal.string(to_ts(time_column)), to=time_column_type)
```

Comment:
Did these change due to the `Nullable` refactor?

Reply:
They were initially made to accommodate the old behavior, but the rest of those changes were reverted after the nullable refactor.

I left these changes in because `.is_type(*LIST)` seems more correct and safer than `.this in LIST`. (It seems like we should avoid reaching into an node's `.this` unless there's no alternative.)